File: backend/src/main/java/com/emerchantpay/backend/domain/transaction/TransactionStatus.java
package com.emerchantpay.backend.domain.transaction;

public enum TransactionStatus {
	TRANSACTION_APPROVED(true),
	TRANSACTION_REVERSED(false),
	TRANSACTION_REFUNDED(true),
	TRANSACTION_ERROR(false);

	private final boolean allowReference;

	TransactionStatus(boolean allowReference) {
		this.allowReference = allowReference;
	}

	public boolean isAllowReference() {
		return allowReference;
	}
}

File: client/src/app/model/transaction/TransactionStatus.ts
class TransactionStatusDef {
  TRANSACTION_APPROVED: Item = {name: 'Approved'};
  TRANSACTION_REVERSED: Item = {name: 'Reversed'};
  TRANSACTION_REFUNDED: Item = {name: 'Refunded'};
  TRANSACTION_ERROR: Item = {name: 'Error'};
}

export type TransactionStatus = keyof TransactionStatusDef;
export const transactionStatus = new TransactionStatusDef();
export const transactionStatuses = Object.keys(transactionStatus) as TransactionStatus[];

interface Item {
  name: string;
}

